modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/OpcUaProtocolAdapter.java

```diff
@@ -97,10 +87,10 @@ public class OpcUaProtocolAdapter implements ProtocolAdapter, WritingProtocolAda
     private final @NotNull ProtocolAdapterState protocolAdapterState;
     private final @NotNull ModuleServices moduleServices;
     private final @NotNull ProtocolAdapterMetricsService protocolAdapterMetricsService;
-    private @Nullable OpcUaClient opcUaClient;
+    private volatile @Nullable OpcUaClient opcUaClient;
```

Comment:
Any chance we can extract client creation and have it final @NotNull ? mutable fields that are used through-out the class are super racy pre se.

Reply:
Done